internal/integration/initramfs_test.go

```diff
@@ -0,0 +1,203 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//go:build integration
+
+package integration_test
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/siderolabs/gen/optional"
+	"github.com/siderolabs/gen/xslices"
+	"github.com/siderolabs/talos/pkg/machinery/extensions"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/u-root/u-root/pkg/cpio"
+	"github.com/ulikunitz/xz"
+	"gopkg.in/yaml.v3"
+)
+
+type initramfsSpec struct {
+	earlyPaths      []string
+	extensions      []string
+	modulesDepMatch optional.Optional[string]
+	flavorID        string
+}
+
+func eatPadding(t *testing.T, in *bufio.Reader) {
+	t.Helper()
+
+	for {
+		b, err := in.ReadByte()
+		if err == io.EOF {
+			return
+		}
+
+		require.NoError(t, err)
+
+		if b != 0 {
+			require.NoError(t, in.UnreadByte())
+
+			return
+		}
+	}
+}
+
+func assertInitramfs(t *testing.T, path string, expected initramfsSpec) {
+	t.Helper()
+
+	f, err := os.Open(path)
+	require.NoError(t, err)
+
+	in := bufio.NewReader(f)
+
+	r := &discarder{r: in}
+
+	if expected.earlyPaths != nil {
+		// first section should be uncompressed
+		var actualEarlyPaths []string
+
+		cpioR := cpio.Newc.Reader(r)
+
+		for {
+			record, err := cpioR.ReadRecord()
+			if err == io.EOF {
+				break
+			}
+
+			require.NoError(t, err)
+
+			if record.Name == "." {
+				continue
+			}
+
+			actualEarlyPaths = append(actualEarlyPaths, record.Name)
+		}
+
+		for _, expectedPath := range expected.earlyPaths {
+			assert.Contains(t, actualEarlyPaths, expectedPath)
+		}
+
+		eatPadding(t, in)
```

Comment:
ahh so there's padding indeed at start of each section?

Reply:
I think `cpio` adds it, even though I can't find a word about it, but kernel supports it, so we do same